releasenotes/notes/deprecate-initial-states-61cd9f2842971ccb.yaml

```diff
@@ -0,0 +1,24 @@
+---
+features:
+  - |
+    The ``HartreeFock`` and ``VSCF`` initial state components are now circuits, 
+    so they provide all methods and flexibility of the ``QuantumCircuit`` object.
+deprecations:
+  - |
+    The ``InitialState`` object and derived classes, as well as the ``VarFormBased``
+    class, have been deprecated. Instead, the ``QuantumCircuit`` can be used.
+    The individual classes emit deprecation warnings on how they can be replaced, 
+    but here is a short overview.
+
+    .. code-block::
+
+        InitialState component | How to replace
+        -----------------------+----------------------------------------------------------------
+        Zero                   | an empty QuantumCircuit
+        Custom(circuit)        | use the circuit directly as input 
+        Custom(state_vector)   | qc = QuantumCircuit(n) 
+                               | qc.initialize(state_vector, qc.qubits)  # normalize vector! 
+        Custom('random')       | use the initialize instruction with random amplitudes 
+        VarFormBased           | use the circuit of the variational form directly 
```

Comment:
We might need to provide some guidance somewhere on this due to the parameter sorting that is done by VQE? As long as you get the optimal dictionary I guess one is good - get the optimal params and you need to take care over sorting. I'm thinking of the use case where we take the var form that had been evolved somewhat by VQE and use that as the starting state for QPE - i.e. the test_vqe2iqpe that I see has been updated to do the above. Maybe we can show this in some tutorial as I am not sure there is a good place in the docs unless we did something in VQE